module-6/src/middlewares/isAdmin.middlware.ts

```diff
@@ -0,0 +1,15 @@
+import { Request, Response, NextFunction } from "express";
+
+export function isAdmin(req: Request, res: Response, next: NextFunction) {
+  const currentUser = req.user;
+
+  if (currentUser?.role !== "admin") {
+    return res.status(403).send({
+      data: null,
+      error: {
+        message: "Only admins can do it",
+      },
+    });
```

Comment:
You can abstract these error responses into a function to reduce duplication and enhance maintainability. You have similar code in Authentication Middleware